### PR TITLE
Importer: XML fields and document matching improvements

### LINKED ISSUE #670: Importer: final feedback before release
### **General comment on the matching** 
We notice that the matching behaviour has high risks of creating duplicates or not deleting records because it is too strict and doesn't recognise records. The issue is mainly with the edition field: 

**Additions:** 

Ex.  https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/213
Line 2: 
Existing record:  https://sandbox-cds-ils.web.cern.ch/backoffice/documents/ayb8t-8hp58 
New record has been created: https://sandbox-cds-ils.web.cern.ch/backoffice/documents/f3v3d-qn214 
=> The difference seems to be the edition.  It did not even create a "partial match". 

Ex. https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/206
Line 5: 
Existing record: https://sandbox-cds-ils.web.cern.ch/backoffice/documents/9nvac-88180
New record: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/206
=> The difference seems to be the edition. 

Proposed solution: If the edition is empty on one side, the edition shouldn't be taken into account in the matching. Or if too complex: completely ignore the edition in the matching.  

**Deletions:** 

https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/209
Line 1 did not match, but the record existed: 
https://sandbox-cds-ils.web.cern.ch/literature/4qj8x-nsc47

### **Conversions remaining issues:**

**Springer**

```
<datafield tag="776" ind1="0" ind2="8">
      <subfield code="i">Printed edition:</subfield>
      <subfield code="z">9783030726713</subfield>
</datafield>
```
-> still not added in the record in Identifier, scheme:ISBN, Material:Print version
There was a confusion in the way the indicators are displayed in my xml reader (Bracket) for those files (ind2 before ind1), that is maybe why my request before was incorrectly written. 

Ex. 
https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/208 ->  https://sandbox-cds-ils.web.cern.ch/backoffice/documents/6mnbz-azb42
In the original file, we have this: 
```
<datafield tag="776" ind2="8" ind1="0">
      <subfield code="i">Printed edition:</subfield>
      <subfield code="z">9789811587016</subfield>
</datafield>
<datafield tag="776" ind2="8" ind1="0">
      <subfield code="i">Printed edition:</subfield>
<subfield code="z">9789811587023</subfield>

```
```
<datafield tag="024" ind1="7" ind2=" ">
      <subfield code="a">10.1007/978-981-16-2086-7</subfield>
      <subfield code="2">doi</subfield>
 </datafield>
```
-> 0247_ still not added in the document record in Identifier, scheme:DOI, Material:Digital (Maybe same issue as before)

See example file: [SpringerNature_Books_Metadata_20211116_013617.zip](https://github.com/CERNDocumentServer/cds-ils/files/7546747/SpringerNature_Books_Metadata_20211116_013617.zip)

**O'reilly**

https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/210
str: Lossy conversion: {'24514c', '24514h'}
-> All indicators should be taken into account (already requested)

ISBNs are not correctly imported (this issue was not noted before):  
`<marc:datafield tag="020" ind1=" " ind2=" "><marc:subfield code="z">9781952157561</marc:subfield></marc:datafield> `
-> Identifier, scheme:ISBN, Material:Print version
`<marc:datafield tag="024" ind1="8" ind2=" "><marc:subfield code="a">9781952157578</marc:subfield></marc:datafield> `
-> Identifier, scheme:ISBN, Material:Digital

See example file: 
[safari-new.xml.zip](https://github.com/CERNDocumentServer/cds-ils/files/7546755/safari-new.xml.zip)

### PR DESCRIPTION
**Pull request by NRodriguezcuellar**
The import now ignores the edition value for document matching if one of the documents does not have an edition value. The ISBN en DOI are now being correctly recognized and pushed to the output document for providers O'reilly and Springer. Added some ignored fields to Safari.

closes #670

### CODE REVIEW

File: cds_ils/importer/providers/safari/rules/document.py [hunk 1]
```diff
@@ -96,12 +96,29 @@ def eitem(self, key, value):
...
         "scheme": "ISBN",
         "value": clean_val("z", value, str, req=True),
+        "material": "PRINT_VERSION"
+    }
+    if isbn not in _identifiers:
+        _identifiers.append(isbn)
+    return _identifiers
+
+
+@model.over("identifiers", "^0248_")
```
Comment (kprzerwa) [inline cds_ils/importer/providers/safari/rules/document.py:112]: in the issue was ind1=7 is 0248 intentional?
```suggestion
@model.over("identifiers", "^0247_")
```

File: cds_ils/importer/providers/springer/rules/document.py [hunk 2]
```diff
@@ -298,7 +298,7 @@ def id_isbns(self, key, value):
     """Translate identifiers isbn."""
     _identifiers = self.get("identifiers", [])
 
-    isbn_value = clean_val("a", value, str)
+    isbn_value = clean_val("a", value, str) or (clean_val('z', value, str))
```
Comment (kprzerwa) [inline cds_ils/importer/providers/springer/rules/document.py:301]: if both subfields will be missing, we will have isbn_value = False

File: cds_ils/importer/providers/safari/rules/document.py (hunk 1 above)
Comment (NRodriguezcuellar) [reply cds_ils/importer/providers/safari/rules/document.py:112]: that was for Springer, this for the last 2 items on the original issue

File: cds_ils/importer/providers/springer/rules/document.py (hunk 2 above)
Comment (NRodriguezcuellar) [reply cds_ils/importer/providers/springer/rules/document.py:301]: it will just be skipped in the next if statement, that is correct behaviour right?

**Review by kprzerwa — APPROVED** [review]

**Pull request merged**